examples/s3express/main.tf

```diff
@@ -0,0 +1,156 @@
+  type        = string
+  sensitive   = true
+}
+
+provider "aws" {
+  region = local.region
+}
+
+# Creating a VPC for this example, you can bring your own VPC
+# if you already have one and don't need to use the one created here.
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "5.18.1"
+
+  name = local.name
+
+  cidr = "10.0.0.0/16"
+
+  azs             = ["${local.region}a", "${local.region}b", "${local.region}c"]
+  private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets  = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+
+  enable_nat_gateway     = true
+  single_nat_gateway     = false
+  one_nat_gateway_per_az = true
+
+  # Default security group in the VPC to allow all egressing.
+  default_security_group_egress = [
+    {
+      description = "Allow all egress"
+      cidr_blocks = "0.0.0.0/0"
+    },
+  ]
+}
+
+# It is highly recommended to create a S3 Gateway endpoint in your VPC.
+# This is prevent S3 network traffic from egressing over your NAT Gateway and increasing costs.
+module "endpoints" {
+  source  = "terraform-aws-modules/vpc/aws//modules/vpc-endpoints"
+  version = "5.18.1"
+
+  vpc_id = module.vpc.vpc_id
+
+  create_security_group      = true
+  security_group_name_prefix = "${local.name}-vpc-endpoints-"
+  security_group_description = "VPC endpoint security group"
+
+  # Security group for the endpoints.
+  # We are allowing everything in the VPC to connect to the S3 endpoint.
+  security_group_rules = {
+    ingress_https = {
+      description = "HTTPS from VPC"
+      cidr_blocks = [module.vpc.vpc_cidr_block]
+    }
+  }
+
+  endpoints = {
+    s3 = {
+      service         = "s3"
+      service_type    = "Gateway"
+      route_table_ids = module.vpc.private_route_table_ids
+      tags            = { Name = local.name }
+    },
+    # Used for S3 Express for lower latency configurations
+    # Ref: https://docs.warpstream.com/warpstream/byoc/advanced-agent-deployment-options/low-latency-clusters
+    s3express = {
+      service         = "s3express"
+      service_type    = "Gateway"
+      route_table_ids = module.vpc.private_route_table_ids
+      tags            = { Name = local.name }
+    }
+  }
+}
+
+# Creating a security group to allow things in the VPC
+# to connect to the WarpStream Agents.
+resource "aws_security_group" "warpstream-connect" {
+  name        = "${local.name}-connect"
+  description = "Allow things the VPC to connect to the WarpStream Kafka Port"
+  vpc_id      = module.vpc.vpc_id
+}
+
+resource "aws_vpc_security_group_ingress_rule" "allow_agent_to_agent_http" {
+  security_group_id = aws_security_group.warpstream-connect.id
+  description       = "Allow Server to Agent Kafka Communication"
+
+  ip_protocol = "tcp"
+  from_port   = 9092
+  to_port     = 9092
+  cidr_ipv4   = "10.0.0.0/16"
+}
+
+# Store the WarpStream Agent Key in AWS Secret Manager
+resource "random_id" "warpstream_agent_key" {
+  byte_length = 4
```

Comment:
what is this?

Reply:
Yea it's just an example. aws secrets have a time destroy so if you do a `tf destroy` and then a `tf apply` again it'll fail because the secret is in a pending destroy, so this just adds a random id to the name. I can add a comment explaining.

Reply:
actually, found out there's a name prefix, so will use that instead.